gateway/src/main/java/com/hurence/webapiservice/http/grafana/GrafanaApiImpl.java

```diff
@@ -116,6 +131,92 @@ at initialization (did not successfully reproduced this in a unit test).//TODO
                 }).subscribe();
     }
 
+    @Override
+    public void export(RoutingContext context) {
+        final long startRequest = System.currentTimeMillis();
+        try {
+            JsonObject requestBody = context.getBodyAsJson();
+            int maxDataPoints = requestBody.getInteger(MAX_POINT_REQUEST_FIELD);
+            if (maxDataPointsAllowedForExportCsv < maxDataPoints ) {
+                throw new IllegalArgumentException("max data points is bigger then allowed");
+            }
+        } catch (Exception ex) {
```

Comment:
Ici il faut séparer le cas ou la query n'a pas pu être parser => BAD_REQUEST du cas ou le nombre de point demander est trop grand => PAYLOAD_TOO_LARGE

Reply:
done, i have separeted the two cases